hibernate-core/src/main/java/org/hibernate/sql/ast/spi/AbstractSqlAstTranslator.java

```diff
@@ -6017,19 +6018,24 @@ public void visitTableReferenceJoin(TableReferenceJoin tableReferenceJoin) {
 	public void visitColumnReference(ColumnReference columnReference) {
 		final String dmlTargetTableAlias = getDmlTargetTableAlias();
 		if ( dmlTargetTableAlias != null && dmlTargetTableAlias.equals( columnReference.getQualifier() ) ) {
-			// todo (6.0) : use the Dialect to determine how to handle column references
-			//		- specifically should they use the table-alias, the table-expression
-			//			or neither for its qualifier
-
-			final String tableExpression = getCurrentDmlStatement().getTargetTable().getTableExpression();
-			// Qualify the column reference with the table expression only in subqueries
-			final boolean qualifyColumn = !queryPartStack.isEmpty();
+			final ColumnQualifierSupport qualifierSupport = getDialect().getMinimumColumnQualifierSupport();
```

Comment:
The changes in the third commit with the introduction of `ColumnQualifierSupport` were mainly needed for Oracle, since with that dialect table aliases are always rendered in the from clause (even for update and delete statements) and we were qualifying the `@Where` clause columns with the table name, which caused an unknown column error.

A simpler `Dialect.supportsUnqualifiedColumns()`, returning `true` by default and `false` for Oracle, could've been enough to solve this but since I noticed this `todo (6.0)` I tried implementing the full feature. This way, we could support dialects *requiring* explicit table expression, even though right now none of the supported ones do AFAIK.